clients/fides-js/src/lib/cookie.ts

```diff
@@ -252,30 +258,75 @@ export const buildCookieConsentForExperiences = (
  *
  * Returns updated experience with user preferences.
  */
-export const updateExperienceFromCookieConsent = (
-  experience: PrivacyExperience,
-  cookie: FidesCookie,
-  debug: boolean
-): PrivacyExperience => {
-  if (!experience.privacy_notices) {
-    return experience;
+export const updateExperienceFromCookieConsent = ({
+  experience,
+  cookie,
+  debug,
+}: {
+  experience: PrivacyExperience;
+  cookie: FidesCookie;
+  debug?: boolean;
+}): PrivacyExperience => {
+  const noticesWithConsent = experience.privacy_notices?.map((notice) => {
+    const preference = Object.hasOwn(cookie.consent, notice.notice_key)
+      ? transformConsentToFidesUserPreference(
+          Boolean(cookie.consent[notice.notice_key]),
+          notice.consent_mechanism
+        )
+      : undefined;
+    return { ...notice, current_preference: preference };
+  });
+
+  // Handle the TCF case, which has many keys to query
+  const tcfEntities = {
+    tcf_purposes: experience.tcf_purposes,
+    tcf_special_purposes: experience.tcf_special_purposes,
+    tcf_features: experience.tcf_features,
+    tcf_special_features: experience.tcf_special_features,
+    tcf_vendors: experience.tcf_vendors,
+    tcf_systems: experience.tcf_systems,
+  };
+
+  if (cookie.tcfConsent) {
+    TCF_COOKIE_KEY_TO_EXPERIENCE_KEY.forEach(({ cookieKey, experienceKey }) => {
```

Comment:
this is a little painful because we have a `PrivacyExperience` object (what we get from the server), a `PrivacyPreferencesRequest` object (what we save to the server), and now also a cookie object. `PrivacyExperience` has one set of keys (`tcf_purposes`, `tcf_features`, etc) and `PrivacyPreferencesRequest` has another set of keys (`purpose_preferences`, `feature_preferences`, etc). I chose to make the cookie object follow the `PrivacyPreferencesRequest` keys, so the cookie looks like

```js
tcfConsent: {
  purpose_preferences: { 1: true }
}
```

because the `PrivacyExperience` keys and the `PrivacyPreferencesRequest` keys don't map to each other, I added this extra `TCF_COOKIE_KEY_TO_EXPERIENCE_KEY` array which essentially defines the mapping. Convincing TS the keys do work is not super fun—the alternative is to do each mapping individually, but then we'd have quite a bit of repeated code. If we wanted the cookie to be a separate key (i.e. `purposes` instead of `purpose_preferences`) we'd need another mapping.

We should probably decide if this cookie key looks okay though, since it is an API of sorts that a user could interact with